lighthouse-core/audits/dobetterweb/bloated-libraries.js

```diff
@@ -0,0 +1,86 @@
+ */
+
+/**
+ * @fileoverview This audit checks a page for any unnecessarily large npm dependencies.
+ * These "bloated" libraries can be replaced with functionally equivalent, smaller ones.
+ */
+
+'use strict';
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when no known "bloated" libraries are detected on the page.*/
+  title: 'Avoids unnecessarily large JavaScript dependencies',
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when some known "bloated" libraries are detected on the page.*/
+  failureTitle: 'Includes unnecessarily large JavaScript dependencies',
+  /** Description of a Lighthouse audit that tells the user why they should care about the large Javascript libraries that have better alternatives.. This is displayed after a user expands the section to see more. No character length limits. */
+  description: 'Large JavaScript libraries can lead to poor performance. ' +
+    'Prefer smaller, functionally equivalent libraries to reduce your bundle size.' +
+    '[Learn more](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#optimize_dependencies',
+  /** Label for a column in a data table. Entries will be names of suggested libraries that are an alternative to what's currently being used. */
+  suggestion: 'Suggestion',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+const database = {'moment': 'dayjs'};
+
+class BloatedLibrariesAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'bloated-libraries',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['Stacks'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {Array<{original: {name: string}, suggestion: {name: string}}>} */
+    const libraryPairings = [];
+    const foundLibraries = artifacts.Stacks.filter(stack => stack.detector === 'js');
+
+    for (const library of foundLibraries) {
+      if (!library.npm || !database[library.npm]) continue;
+      libraryPairings.push({original: library, suggestion: {name: database[library.npm]}});
+    }
+
+    const tableDetails = [];
+    libraryPairings.forEach(libraryPairing => {
```

Comment:
protip: if you use `const tableDetails = libraryPairings.map(...)` then typescript will be able to infer the type of `tableDetails` and won't fallback to `any[]` :)

Reply:
It's still unable to infer :/ should I manually comment the type?

Reply:
hm, is something up with your intellisense then? it should definitely infer the type

![image](https://user-images.githubusercontent.com/2301202/87480651-6165ae80-c5f3-11ea-8eda-f0288ab4eb89.png)

Reply:
Nvm yeah works now